### PR TITLE
fix: fix numerical issue in Seidel 1D solver.

### PR DESCRIPTION
**Pull request by jmirabel**
Rework handling of numerical issues to a more theoretically sound method.

Checklists:
- [x] Update HISTORY.md with a single line describing this PR

### CODE REVIEW

File: cpp/src/toppra/solver/seidel-internal.hpp [hunk 1]
```diff
@@ -84,19 +86,24 @@ LpSol1d solve_lp1d(const RowVector2& v, const Eigen::MatrixBase<Derived>& A)
...
+    // a[i] -> a and b[i] -> b in this comment
+    //
+    // case a == 0.0:
+    //   feasible iif b < ABS_TOLERANCE (tolerate a small violation)
+    // otherwise
+    //   if a>0 b<0 then x <= +|b/a| so handled as a normal constraint
+    //   if a<0 b<0 then x >= -|b/a| so handled as a normal constraint
+    //
+    //   if a>0 b>0 then x <= -|b/a| so infeasible if |b/a| > 1/REL_TOLERANCE
+    //   if a<0 b>0 then x >= +|b/a| so infeasible if |b/a| > 1/REL_TOLERANCE
```
Comment (jmirabel) [inline cpp/src/toppra/solver/seidel-internal.hpp:99]: See this comment in the code to know how numerical issues are now handled.

Comment (hungpham2511) [reply cpp/src/toppra/solver/seidel-internal.hpp:99]: It's implicit (to me at least) that x is not allowed to be outside of - 1 / REL and 1 / REL range, yes?

If so please document it.

Comment (hungpham2511) [reply cpp/src/toppra/solver/seidel-internal.hpp:99]: Otherwise the logic looks fine to me.

**Review by hungpham2511 — APPROVED** [review]
LGTM

File: cpp/src/toppra/solver/seidel-internal.hpp (hunk 1 above)
Comment (jmirabel) [reply cpp/src/toppra/solver/seidel-internal.hpp:99]: The fact that it is not allowed is a consequence of the logic but we can indeed state it.

Comment (jmirabel) [reply cpp/src/toppra/solver/seidel-internal.hpp:99]: Actually, no, `x` can be outside the range you say.
1. When `b<0`, we do not change the constraint.
2. When `b>0`, the problem is said infeasible if `|b/a|` is too large. But if it is not, the constraint is only on one side of `x` and the optimum may actually be on the other direction, which may be bound beyond +/- 1/REL.

**Pull request merged**